instill/model/v1alpha/model.proto

```diff
@@ -4,153 +4,341 @@ package instill.model.v1alpha;
 
 // Protobuf standard
 import "google/protobuf/struct.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
 
 // Google api
-import "google/api/annotations.proto";
+import "google/api/resource.proto";
 import "google/api/field_behavior.proto";
 
 import "instill/model/v1alpha/definition.proto";
 
-// LivenessRequest represents a request to check a service liveness status
-message LivenessRequest {
-  // Service name to check for its liveness status
-  string service = 1 [ (google.api.field_behavior) = OPTIONAL ];
-}
-
-// LivenessResponse represents a response for a service liveness status
-message LivenessResponse {
-  // ServingStatus enumerates the status of a queried service
-  enum ServingStatus {
-    // Serving status: UNSPECIFIED
-    SERVING_STATUS_UNSPECIFIED = 0;
-    // Serving status: SERVING
-    SERVING_STATUS_SERVING = 1;
-    // Serving status: NOT SERVING
-    SERVING_STATUS_NOT_SERVING = 2;
+// Model represents a model
+message Model {
+  option (google.api.resource) = {
+    type: "api.instill.tech/Model"
+    pattern: "models/{model}"
+  };
+
+  // Model visibility including public or private
+  enum Visibility {
+    // Visibility: UNSPECIFIED, equivalent to PRIVATE.
+    VISIBILITY_UNSPECIFIED = 0;
+    // Visibility: PRIVATE
+    VISIBILITY_PRIVATE = 1;
+    // Visibility: PUBLIC
+    VISIBILITY_PUBLIC = 2;
   }
 
-  // Status is the instance of the enum type ServingStatus
-  ServingStatus status = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-}
-
-// ReadinessRequest represents a request to check a service readiness status
-message ReadinessRequest {
-  // Service name to check for its readiness status
-  string service = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Resource name. It must have the format of "models/{model}".
+  // For example: "models/yolov4"
+  string name = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ] ;
+  // Model ID in UUIDv4
+  string id = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model display name, this is used to construct the resource name.
+  string display_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
```

Comment:
@xiaofei-du it should be required when creating a model?

Reply:
We removed `display_name` (also `full_name`). Resource ID `id` and resource name `name` should be enough to identify a model. E.g., a model example in response will have 
```
{
  "name": "models/yolov4",
  "id": "yolov4",
  ...
}
```

We also made `id` [`IMMUTABLE`](https://google.aip.dev/203#immutable). Therefore, `id` of a model is set during creation with `POST`, but can't be updated with `PATCH`.

To rename a model (update `id` of a model), use the custom `rename` method.

See guide about [`PATCH`](https://cloud.google.com/apis/design/standard_methods#update)
> Any functionality to rename or move a resource must not happen in the Update method and instead shall be handled by a custom method.